analyzers/src/SonarAnalyzer.CSharp/Rules/LiteralsShouldNotBePassedAsLocalizedParameters.cs

```diff
@@ -103,22 +104,27 @@ private static void AnalyzeAssignments(SyntaxNodeAnalysisContext context)
             }
         }
 
-        private static bool IsStringLiteral(ExpressionSyntax expression, SemanticModel semanticModel) =>
+        private static bool IsStringLiteral(SyntaxNode expression, SemanticModel semanticModel) =>
             expression != null
-            && semanticModel.GetConstantValue(expression) is { HasValue: true } constant
-            && constant.Value is string;
+            && semanticModel.GetConstantValue(expression) is { HasValue: true, Value: string _ };
 
         private static bool IsLocalizable(ISymbol symbol) =>
             symbol?.Name != null
-            && symbol.GetAttributes(KnownType.System_ComponentModel_LocalizableAttribute) is var localizableAttribute
-            && IsLocalizable(symbol.Name, localizableAttribute);
+            && symbol.GetAttributes(KnownType.System_ComponentModel_LocalizableAttribute) is var localizableAttributes
+            && IsLocalizable(symbol.Name, new List<AttributeData>(localizableAttributes).AsReadOnly());
 
-        private static bool IsLocalizable(string symbolName, IEnumerable<AttributeData> localizableAttribute) =>
-            localizableAttribute.Any(x => AttributeHasConstructorArgument(x, true))
+        private static bool IsLocalizable(string symbolName, IReadOnlyCollection<AttributeData> localizableAttributes) =>
+            localizableAttributes.Any(x => HasConstructorWithBoolValue(x, true))
             || (symbolName.SplitCamelCaseToWords().Any(LocalizableSymbolNames.Contains)
-               && (!localizableAttribute.Any(x => AttributeHasConstructorArgument(x, false))));
+               && (!localizableAttributes.Any(x => HasConstructorWithBoolValue(x, false))));
+
+        private static bool IsLocalizableStringLiteral(IParameterSymbol parameter, ArgumentSyntax argumentSyntax, SemanticModel semanticModel) =>
+            parameter != null
+            && argumentSyntax != null
+            && IsLocalizable(parameter)
+            && IsStringLiteral(argumentSyntax.Expression, semanticModel);
 
-        private static bool AttributeHasConstructorArgument(AttributeData attribute, bool expectedValue) =>
+        private static bool HasConstructorWithBoolValue(AttributeData attribute, bool expectedValue) =>
```

Comment:
We can rename it to `HasConstructorWitValue`. `expectedValue` is already `bool` so we don't need to mention it in the method name. What do you think?